core/drivers/crypto/hisilicon/include/hpre_dh.h

```diff
@@ -0,0 +1,31 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, HiSilicon Technologies Co., Ltd.
+ */
+
+#ifndef __TEE_HPRE_DH_H__
+#define __TEE_HPRE_DH_H__
+
+#include <drvcrypt.h>
+#include <drvcrypt_acipher.h>
+
+#define HPRE_DH_MAX_KEY_BYTES 512
+#define HPRE_DH_TOTAL_BUF_SIZE(key_bytes) ((key_bytes) * 4)
+struct hpre_dh_msg {
+	uint8_t *x_p; /* X and p data in order */
+	uintptr_t x_p_dma;
```

Comment:
If this is an integer representing a virtual address, please use vaddr_t. If it's a physical address, use paddr_t. Same below.